Backend/app/rider-platform/rider-app/Main/src/Domain/Types/HotSpot.hs

```diff
@@ -0,0 +1,59 @@
+module Domain.Types.HotSpot where
+
+import qualified Data.Geohash as DG
+import Data.Text
+import Kernel.External.Maps
+import Kernel.Prelude
+import Kernel.Types.Common hiding (id)
+import Kernel.Utils.Common
+import Tools.Metrics
+
+data HotSpot = HotSpot
+  { geoHash :: Maybe Text,
+    frequency :: Int,
+    centroidLatLong :: Maybe LatLong
+  }
+  deriving (Generic, ToJSON, FromJSON, ToSchema)
+
+data HotSpotResponse = HotSpotResponse
+  { hotSpots :: [HotSpot],
+    blockRadius :: HighPrecMeters
+  }
+  deriving (Generic, ToJSON, FromJSON, ToSchema)
+
+convertToHotSpot ::
+  ( CoreMetrics m,
+    HasFlowEnv m r '["hotSpotGeoHashPrecision" ::: Int]
+  ) =>
+  LatLong ->
+  m HotSpot
+convertToHotSpot LatLong {..} = do
+  geoHashLength <- asks (.hotSpotGeoHashPrecision)
```

Comment:
lets move these configs to DB

Reply:
name of the db?

Reply:
how can I get merchant id here?

onUpdate ValidatedRideAssignedReq {..} =